edx/code-annotations
# Stub implementation of static annotation search and plaintext extension

**Description:** A skeleton implementation of static annotation configuration, search, reporting, and extensions.

**JIRA:** PLAT-2350, PLAT-2361, PLAT-2365

**Installation instructions:** To test with the extensions you will need to install the package with `pip install --editable .`

**Testing instructions:**

0. Make sure the package is installed as above
1. Add some annotations to various files
2. Use the `.annotations_sample` as a template to create your own config file, or just point to it on the command line if you're using the `.. pii::` annotations
3. Run `find_annotations`. If things are misbehaving try the various levels of verbosity (`-v` - `-vvv`)

**Merge checklist:**

**Post merge:**
      finished.

**Author concerns:** 
- There are a lot of TODO items, mostly things to be addressed in other tickets in this epic
- There are no tests currently as those should be done with the tickets to flesh out this skeleton
- The current implementation of the `plaintext` extension started out as the `python` one, but can be used generically. If we want to limit the Python extension to just Python comments we can build on this, but this was in a good enough state to merge as an example.
